src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -137,8 +138,23 @@ export class HuiCardPicker extends LitElement {
       return;
     }
 
-    this._unusedEntities = computeUnusedEntities(this.hass, this.lovelace);
-    this._usedEntities = [...computeUsedEntities(this.lovelace)];
+    this._unusedEntities = computeUnusedEntities(
```

Comment:
`computeUnusedEntities` calculates `computeUsedEntities` under the hood. This means that we're now calculating that twice and it's expensive. We should refactor it so we have 1 function to calculate the diff that we can use here.

Reply:
Not sure what you mean by calculate the difference

Reply:
refactor it so we call a new function that returns both or refactor `computeUnusedEntites` to accept the used entities as a param so it doesn't calc again

Reply:
So we calculate unused entities using `used entities - entities for which we have states availabale`. So when we compute unused entities and then compute used entities, we are computing used entities twice. We should break up the `computeUnusedEntities` function in 2. One that just computes the difference and one that calls computeUsedEntities and passes it to the calc function.

Reply:
Honestly, I feel like we could do this a lot easier if everything was sets, so we can just remove 1 set from another.

Reply:
I have updated the functions and added a calc function. Also using sets more now. not sure the full extent you would like to take it. 

Also, I didn't find an easy way to just `remove 1 set from another`